lib/linguist/heuristics.yml

```diff
@@ -340,6 +340,10 @@ disambiguations:
   - language: C++
     named_pattern: cpp
   - language: C
+- extensions: ['.c3', '.c3i', '.c3t']
+  rules:
+  - language: C3
+    named_pattern: c3
```

Comment:
A heuristic is not necessary when there is only one language using an extension. You can remove this.

Reply:
There are [1] [2] [3] some that have the C3 extension (see https://github.com/search?q=%28path%3A*.c3+OR+path%3A*.c3i+OR+path%3A*.c3t%29+language%3AC+AND+NOT+owner%3Ac3lang+NOT+is%3Afork&type=code)
[1]: https://github.com/windelbouwman/ppci/blob/915c069e0667042c085ec42c78e9e3c9a5295324/librt/io.c3#L4
[2]: https://github.com/arcanon/ipadflash/blob/6e3f35fe4b36156d79d3b09ac9b13f36a9802346/src/SDL/src/main/dummy/SDL_dummy_main.c3#L4
[3]: https://github.com/as112/STM32F103-Boiler-Controller/blob/7d84ac6992df5cbe436a73450e2d2a10eb6fb8c7/ds18b20-termostat-v2/CMSIS/src/core_cm3.c3#L4

Reply:
The heuristics should help.
Also, should help more in cases where there are no file extension, if linguist even considers them or uses the heuristics / or samples.

Reply:
The heuristic won't help at all as it won't be used. If linguist only knows of one language with a certain extension, it'll identify all files it finds with that extension as that language

Reply:
Oh, ok.